### PR TITLE
restore: merge tidb-tools/pkg/restore-util

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

This PR merges tidb-tools/pkg/restore-util into br, it eases maintenance burden.

It also changes few code in order to pass linters, please review https://github.com/pingcap/br/pull/146/commits/23aa6f72b8f3c5a0f47c090b38a961dd8d764383 .

### Check List 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -9,7 +9,8 @@ issues:
       text: "Potential HTTP request made with variable url"
       linters:
         - gosec
-    - path: .go
-      text: "Use of weak random number generator"
+    # TODO Remove it.
+    - path: client.go
```
Comment (kennytm) [inline .golangci.yml]:
```suggestion
    - path: split_client.go
```

File: pkg/restore/split_client.go [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+		SplitKey: key,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if resp.RegionError != nil {
+		return nil, errors.Errorf("split region failed: region=%v, key=%x, err=%v", regionInfo.Region, key, resp.RegionError)
+	}
+
+	// BUG: Left is deprecated, it may be nil even if split is succeed!
```
Comment (overvenus) [inline pkg/restore/split_client.go]: PTAL @5kbpers

Comment (5kbpers) [reply pkg/restore/split_client.go]: It would be found in the `Regions` if resp (see L156~L164)

**Comment by kennytm** [comment]
:shipit:

**Pull request merged**